src/common/cups_print.h

```diff
@@ -59,15 +59,12 @@ typedef struct dt_print_info_t
   dt_paper_info_t paper;
 } dt_print_info_t;
 
-// is some printers are available
-gboolean is_printer_available(void);
+// Asynchronous printer discovery, cb will be called for each printer found
```

Comment:
That's not how I read your code and the cups api docs. Both cupsEnumDests and cupsGetDests only return when they have fully enumerated the printers. So your current code won't block the app (because it's run in a background job) but it won't show up any printers until all printers are enumerated. What cupsEnumDests does differently than cupsGetDests is that it will call a callback every time a printer is enumerated, so while it will only still return after all are enumerated it will call a callback of your choice on every printer while it's doing it. If you make that callback add a new printer you'll have available printers right away even though cupsEnumDests has not finished yet.

Either this solution or some other one exists because the gnome printer dialog is capable of listing printers and stay responsive even before being able to show all printer options. Our best bet is to find a way to generate a test case, probably by creating a network printer that doesn't exist so that cups tries to connect to it until timeout.